loading.bs

```diff
@@ -331,6 +386,391 @@ Note: Applying the signed exchange's response here has the effect of letting a
+    [=response/header integrity value=] of the [=alternate signed exchange=],
+    encoded as a [[CSP]] <a grammar>hash-source</a>. 
+* An <dfn>alternate signed exchange link</dfn> is a <a http-header>Link</a>
+    header sent with a signed exchange |S|, with the <{link/rel/alternate}> link
+    type, the `type` parameter set to `application/signed-exchange`, and a
+    [=Link Context=] (an `anchor` parameter). The [=Link Context=] MUST be the
+    [=Link Target=] of a <{link/rel/preload}> <a http-header>Link</a> header
+    inside the signed content of |S|, and the [=alternate signed exchange link=]
+    means that this resource can also be found inside the <dfn
+    dfn-type=dfn>alternate signed exchange</dfn> at the [=Link Target=] of the
+    [=alternate signed exchange link=].
+
+* New [=Target Attributes=] on the <a http-header>Link</a> header.
+
+    * The <dfn element-attr for="link">variants</dfn> parameter may be present.
+        This parameter is used in <{link/rel/allowed-alt-sxg}> link and
+        [=alternate signed exchange link=] to indicate what representations are
+        available for the resource at the time that the response is produced.
+        This attribute is using the same format as <a
+        http-header>Variants</a> Header Field.
+
+    * The <dfn element-attr for="link">variant-key</dfn> parameter may be
+        present. This parameter is used in <{link/rel/allowed-alt-sxg}> link and
+        [=alternate signed exchange link=] to indicate one or more sets of
+        available-values that identify the representation of the resource.
+        This parameter is using the same format as <a
+        http-header>Variant-Key</a> Header Field.
+
+<h3 algorithm id="mp-document">New Document fields</h3>
+
+At the end of
+[the Document object](https://html.spec.whatwg.org/multipage/dom.html#the-document-object)
+section, add the following lines:
+
+<blockquote dfn-for="Document">
+
+The {{Document}} has a map of <dfn dfn>prefetched signed exchanges for
+navigation</dfn>, which is a [=map=] of [=URL=] to [=exchange=], initially
+empty.  <span class="note">This might merge with the set of prefetched resources
+when <{link/rel/prefetch}> is specified more completely.</span>
+
+The {{Document}} has a set of <dfn dfn>prefetched
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+</blockquote>
+
+<h3 algorithm id="mp-navigation-params">New navigation params struct field</h3>
+
+In the [=navigation params=], add the following items:
+
+<dl dfn-for="navigation params">
+
+: <dfn>prefetched subresource signed exchanges</dfn>
+:: A set of [=exchanges=], initially empty.
+
+</dl>
+
+<h3 algorithm id="mp-link-type-prefetch">Monkeypatch Link type "prefetch"</h3>
+
+Note: To prefetch the [=alternate signed exchanges=] when the UA receives a
+prefetched main resource signed exchange, this section monkeypatchs
+Link type "<{link/rel/prefetch}>".
```

Comment:
It would be good to make the note more descriptive, outlining that this is getting all the alternateLinks from the outer response, see if they're allowed by the inner response, matches preloads to allowed, etc.